packages/enzyme/src/ShallowWrapper.js

```diff
@@ -1411,7 +1493,15 @@ class ShallowWrapper {
       if (!isCustomComponentElement(el, adapter)) {
         throw new TypeError(`ShallowWrapper::${name}() can only be called on components`);
       }
-      return this.wrap(el, null, { ...this[OPTIONS], ...options });
+      return this.wrap(el, null, {
+        ...this[OPTIONS],
+        ...options,
+        context: {
+          ...this[OPTIONS].context,
+          ...options.context,
```

Comment:
i wonder if we should instead consider the `context` option as *overriding* the proper child context - ie, only do the correct thing when a context option isn't supplied? (or alternatively, not pass forward the context option?

Reply:
I like the idea of the `context` option being an override!